src/socket.rs

```diff
@@ -530,7 +532,11 @@ impl UtpSocket {
         loop {
             // Abort loop if the current try exceeds the maximum number of retransmission retries.
             if retries >= self.max_retransmission_retries {
+                println!("exceeds max_retransmission_retries : {} ; current connect state is : {:?}",
+                         self.max_retransmission_retries, self.state);
                 self.state = SocketState::Closed;
+                println!("socket marked as closed from {:?} to {:?}",
+                         self.local_addr(), self.connected_to);
```

Comment:
This is to highlight the place that is in suspicious, once the work is done, it will be removed or changed to `debug!`